Upstream-status: Pending.

Fix build with Eigen 3.4 by using the runtime jacobiSvd API.

--- a/intern/libmv/libmv/multiview/euclidean_resection.cc
+++ b/intern/libmv/libmv/multiview/euclidean_resection.cc
@@ -107,7 +107,7 @@ void AbsoluteOrientation(const Mat3X &X, const Mat3X &Xp, Mat3 *R, Vec3 *t) {
   N(3, 3) = Sxx + Syy + Szz;
 
   // The first eigenvector is the solution quaternion.
-  Vec4 q = N.jacobiSvd<Eigen::ComputeFullU>().matrixU().col(0);
+  Vec4 q = N.jacobiSvd(Eigen::ComputeFullU).matrixU().col(0);
 
   // Write resulting rotation matrix.
   (*R)(0, 0) = 1 - 2 * q(2) * q(2) - 2 * q(3) * q(3);
@@ -246,7 +246,7 @@ void EuclideanResectionAnsarDaniilidis(const Mat2X &x,
 
   // Get right singular vector corresponding to smallest singular value.
   int rank = M.rows() - M.cols();
-  Mat V = M.jacobiSvd<Eigen::ComputeFullV>().matrixV().block(
+  Mat V = M.jacobiSvd(Eigen::ComputeFullV).matrixV().block(
       0, M.cols() - rank - 1, M.cols(), rank + 1);
 
   // Last but one column of V.
@@ -303,7 +303,7 @@ void EuclideanResectionAnsarDaniilidis(const Mat2X &x,
     // by estimating the nullspace of K.
     int num_k_columns = static_cast<int>(K.cols());
     Vec x3 =
-      K.jacobiSvd<Eigen::ComputeFullV>().matrixV().col(num_k_columns - 1);
+      K.jacobiSvd(Eigen::ComputeFullV).matrixV().col(num_k_columns - 1);
     vector<Vec> ks;
     Nullspace(&K, &ks);
     Vec x4 = ks[0];
